tools/mtouch/Target.cs

```diff
@@ -802,9 +813,31 @@ public void ProcessAssemblies ()
 
 			ManagedLink ();
 
+			if (App.EnableDedup)
+				GenerateDedup ();
+
 			GatherFrameworks ();
 		}
 
+		public void GenerateDedup ()
+		{
+			var dedupDummyDll = String.Format ("{0}.dll", DedupDummyName);
+			// Make available for the AOT compiler
+			var dllPath = Path.Combine (BuildDirectory, dedupDummyDll);
+
+			// Qualified because of collisions with cecil
+			var aName = new System.Reflection.AssemblyName (DedupDummyName);
+			var ab = AppDomain.CurrentDomain.DefineDynamicAssembly (aName, System.Reflection.Emit.AssemblyBuilderAccess.RunAndSave, BuildDirectory);
+
+			// Make .dll file
+			ab.Save (dedupDummyDll);
```

Comment:
Is the dedup assembly needed at runtime? Or only at build time?

Reply:
it's a constant, empty assembly being created each time right ?
maybe we can embed one (pre-created) and extract it when needed (saving build time)

Reply:
We can probably just install it in /Library/Frameworks/Xamarin.iOS.framework and build it directly from there.

One point here is that using reflection like this will reference the desktop profile (v4.5 mscorlib) instead of the correct profile.

Reply:
@rolfbjarne not sure I follow wrt reflection. This looked like an empty assembly where dedup'ed code is moved (shared) later